pkgs/by-name/vp/vpp/package.nix

```diff
@@ -0,0 +1,141 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  nix-update-script,
+  cmake,
+  pkg-config,
+  check,
+  openssl,
+  python3,
+  subunit,
+  mbedtls_2,
+  libpcap,
+  libnl,
+  libmnl,
+  libelf,
+  dpdk,
+  jansson,
+  zlib,
+  rdma-core,
+  libbpf,
+  xdp-tools,
+  enableDpdk ? true,
+  enableRdma ? true,
+  # FIXME: broken: af_xdp plugins - no working libbpf found - af_xdp plugin disabled
+  enableAfXdp ? false,
+}:
+
+let
+  dpdk' = dpdk.overrideAttrs (old: {
+    mesonFlags = old.mesonFlags ++ [ "-Denable_driver_sdk=true" ];
+  });
+
+  rdma-core' = rdma-core.overrideAttrs (old: {
+    cmakeFlags = old.cmakeFlags ++ [
+      "-DENABLE_STATIC=1"
+      "-DBUILD_SHARED_LIBS:BOOL=false"
+    ];
+  });
+
+  xdp-tools' = xdp-tools.overrideAttrs (old: {
+    postInstall = ''
+      # Drop unfortunate references to glibc.dev/include at least from $lib
+      nuke-refs "$lib"/lib/bpf/*.o
```

Comment:
This should preferable fixed in xdp-tools directly.

Reply:
To be completely honest, I pretty much blindly copied this part from [the first PR](https://github.com/NixOS/nixpkgs/blob/eaf95021d2e6beab6f3dfab4fa83031f888e0010/pkgs/servers/vpp/default.nix#L34-L39) to get around a `libxdp.a library not found` error, so until now I haven't looked into why it's necessary, but now that I'm looking into I agree. I'll make another PR, then update VPP once it's merged.

Reply:
I just checked. That line is not needed anymore. `xdp-tools.lib` does not have any references to `glibc.dev`

Reply:
Alright, I'll incorporate that into the PR as well when I get around to making it. For the time being it's still necessary to override `postInstall` to `""` in this package so the static libs aren't removed.

I'll make the PR change xdp-tools' `postInstall` to work similarly to [SDL2's](https://github.com/NixOS/nixpkgs/blob/54cce652cb480258b12f15b2915e7a6c79ac51ae/pkgs/development/libraries/SDL2/default.nix#L185-L189), after which only the `dontDisableStatic` line will be necessary in VPP.

Reply:
I meant: just remove the nuke-refs line here. That is sufficient for now.

There is no need to modify xdp-tools globally. We should not build static libs unless explicitly requested.

Reply:
I think it would be beneficial to make xdp-tools build static libs if requested with `dontDisableStatic = true` (like SDL2 does) instead of having to completely overwrite its `postInstall`. Either way, the change in VPP is implemented, so I think it'd be best to leave this topic be for now and have this discussion in the future PR I'll create.